lib/std/wasm.zig

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const testing = @import("std.zig").testing;
+
+/// Wasm instruction opcodes
+///
+/// All instructions are defined as per spec:
+/// https://webassembly.github.io/spec/core/appendix/index-instructions.html
+pub const Opcode = enum(u16) {
```

Comment:
Opcodes are one byte. This enum should use a byte as its tag type.
This enum should also be made non-exhaustive so that the spec adding new opcodes is not a breaking change.
```suggestion
pub const Opcode = enum(u8) {
```